src/Region/RegionHandler.cc

```diff
@@ -2154,4 +2282,168 @@ casacore::Vector<float> RegionHandler::GetTemporaryRegionProfile(
     return profile;
 }
 
+void RegionHandler::GetStokesPtotal(
+    const ProfilesMap& profiles_q, const ProfilesMap& profiles_u, const ProfilesMap& profiles_v, ProfilesMap& profiles_ptotal) {
+    auto calc_step1 = [&](double q, double u) {
+        if (!std::isnan(q) && !isnan(u)) {
```

Comment:
One more thing I just noticed: why is the second `isnan` not a `std::isnan` in all these functions?